external/java/src/main/java/ch/epfl/dedis/byzcoin/Proof.java

```diff
@@ -112,64 +159,183 @@ public StateChangeBody getValues() {
     /**
      * @return the value of the proof.
      */
-    public byte[] getValue(){
+    public byte[] getValue() {
         return getValues().getValue();
     }
 
     /**
      * @return the string of the contractID.
      */
-    public String getContractID(){
+    public String getContractID() {
         return new String(getValues().getContractID());
     }
 
     /**
      * @return the darcID defining the access rules to the instance.
-     * @throws CothorityCryptoException if there's a problem with the cryptography
      */
-    public DarcId getDarcID() throws CothorityCryptoException{
+    public DarcId getDarcID() {
         return getValues().getDarcId();
     }
 
     /**
-     * @return true if this looks like a matching proof for byzcoin.
+     * Checks if the proof is valid and of type expected.
+     *
+     * @param expected the expected contractId
+     * @param id       the Byzcoin id to verify the proof against
+     * @return true if the proof is correct with regard to that Byzcoin id and the contract is of the expected type.
+     * @throws CothorityException if something goes wrong
      */
-    public boolean isByzCoinProof(){
-        if (!matches()) {
-            return false;
-        }
-        return true;
+    public boolean isContract(String expected, SkipblockId id) throws CothorityException {
```

Comment:
I was confused with the name at first. Should this be renamed to something like _isContractOk_ or better ?

Reply:
Also the comment says the exception is thrown if something goes wrong but verify will throw if the byzcoin id is bad, so nothing went wrong, it's just the id that is wrong. Is it the expected behaviour to have either a boolean or an error ?

Reply:
yes, the name could be improved, maybe to something like `contractIsType` because it's checking whether the contract matches the expected type. regarding the second comment, `verify` should never fail when the system is working correctly (the forward links should always checkout), but it might return false because the client used a wrong contract name, so it's two different types of errors. I will update the comment.